lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -126,6 +126,36 @@ ::mlir::LogicalResult mlir::tt::ttir::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  auto shape = getShape();
+  int64_t shape_size = static_cast<int64_t>(shape.size());
+
+  if (shape_size == 0) {
+    return emitOpError("Shape attribute must be non-empty");
+  }
+
+  if (shape_size > 5) {
+    return emitOpError("Shape attribute must have at most 5 elements");
+  }
+
+  if (shape_size != outputType.getRank()) {
+    return emitOpError("Shape attribute must have the same rank as the output");
+  }
+
+  int index_of_negative_1 = -1;
+  for (int64_t i = 0; i < shape_size; i++) {
+    if (mlir::cast<IntegerAttr>(shape[i]).getInt() == -1) {
+      if (index_of_negative_1 != -1) {
+        return emitOpError("Shape attribute must have at most one -1 element");
+      }
+      index_of_negative_1 = i;
+    }
+  }
```

Comment:
My personal opinion is that it'd be much more readable if we didn't keep track of the index of -1, but the number of occurrences (or just a `bool` to track if we've seen one already). I'll leave it up to you to decide whether to make the change.

Reply:
Makes sense to me sure :)